### PR TITLE
Feature/126 accordion table sort and group no value fields

### PR DESCRIPTION
**Pull request by lestard**
### Prerequisites

Please make sure you can check the following two boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [x] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### Description

See #126. 

- fix sorting of items that don't have a value for the `sortBy`/`groupBy` column
- change order of items so that items without a value are moved to the end of the list
- `AccordionTable`: 
	- add a separate pseudo-group for items that don't have a value for the `groupBy` column
	- add an optional prop to define the label for this new pseudo-group
- change sorting for values of type number so that `0` isn't handled as `falsy` anymore but instead is sorted and grouped like other numbers

Breaking changes: At the moment the handling of Tables with items that don't have a value defined for the `sortBy`/`groupBy` column isn't working correctly so I assume that most people don't have not-defined values in their data. However, as this PR changes the way not-defined values are handled in theory it could change the behaviour of existing apps that depend on the (wrong) sorting and grouping of not-complete items.

### CODE REVIEW

File: src/components/AccordionTable/AccordionTable.types.part.tsx [hunk 1]
```diff
@@ -108,6 +108,12 @@ export interface AccordionTableProps<T> extends TableProps<T>, AccordionCardLabe
    * Notification callback if a group is expanded or collapsed.
    */
   onToggleGroup?(e: AccordionGroupToggleEvent<T>): void;
+
+  /**
+   * A label to be used for the group of those items which have no value for the 'groupBy' column.
+   * Default value is `—`.
```
Comment (kotielnikov) [inline src/components/AccordionTable/AccordionTable.types.part.tsx]: @default '-'

File: src/components/AccordionTable/AccordionTableBasic.part.tsx [hunk 2]
```diff
@@ -186,16 +220,40 @@ export class AccordionTableBasic<T> extends React.Component<AccordionTableProps<
     );
   }
 
+  private getGroupByValue = (rowData: any) => {
```
Comment (kotielnikov) [inline src/components/AccordionTable/AccordionTableBasic.part.tsx]: why `any`?

Comment (lestard) [reply src/components/AccordionTable/AccordionTableBasic.part.tsx]: If I use `T` instead I have the problem with the check `if (rowValue || rowValue === 0)`. TypeScript tells me that `rowValue` won't ever be a number and therefore the expression `rowValue === 0` will always be false.

File: src/components/AccordionTable/AccordionTable.types.part.tsx (hunk 1 above)
Comment (kotielnikov) [reply src/components/AccordionTable/AccordionTable.types.part.tsx]: with "@": `@default: '-'`

Comment (lestard) [reply src/components/AccordionTable/AccordionTable.types.part.tsx]: I'm a little confused now on what the correct syntax is. Would this be correct:
```
  /**
   * A label to be used for the group of those items which have no value for the 'groupBy' column.
   * Default value is `—`.
   *
   * @default '—'
   */
```

Comment (kotielnikov) [reply src/components/AccordionTable/AccordionTable.types.part.tsx]:
```
  /**
   * A label to be used for the group of those items which have no value for the 'groupBy' column.
   *
   * @default '—'
   */
```

**Comment by lestard** [comment]
Hi @dimabory. This is true, the first PR was merged to master by mistake and therefore the changes were not included in newer alpha-releases on npm. I talked to @guiherzog and he suggested to create a new PR targeting `develop` this time. 
Sorry for the trouble.

**Pull request merged**